**Confidence Interval**

In [1]:

#Task:
#- Calculate 99% CI for print-head durability using:
 #   1. Sample SD (t-distribution)
  #  2. Known population SD = 0.2 (z-distribution)


import math
import statistics
from scipy import stats  # for critical values

# Data: durability of print heads (millions of characters)
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48,
        1.20, 1.33, 1.18, 1.22, 1.29]
n = len(data)

# Sample statistics
mean_sample = statistics.mean(data)
sd_sample = statistics.stdev(data)  # sample SD
se_sample = sd_sample / math.sqrt(n)

print("\n--- Confidence Interval using SAMPLE SD (t-distribution) ---")
df = n - 1
alpha = 0.01  # 99% confidence
t_crit = stats.t.ppf(1 - alpha/2, df)  # two-tailed
margin_t = t_crit * se_sample
ci_t = (mean_sample - margin_t, mean_sample + margin_t)

print(f"Sample mean = {mean_sample:.6f}")
print(f"Sample SD = {sd_sample:.6f}")
print(f"t critical (df={df}) = {t_crit:.6f}")
print(f"99% CI = {ci_t}")

print("\n--- Confidence Interval using KNOWN σ = 0.2 (z-distribution) ---")
sigma_pop = 0.2
se_pop = sigma_pop / math.sqrt(n)
z_crit = stats.norm.ppf(1 - alpha/2)
margin_z = z_crit * se_pop
ci_z = (mean_sample - margin_z, mean_sample + margin_z)

print(f"Population σ = {sigma_pop}")
print(f"z critical = {z_crit:.6f}")
print(f"99% CI = {ci_z}")



--- Confidence Interval using SAMPLE SD (t-distribution) ---
Sample mean = 1.238667
Sample SD = 0.193164
t critical (df=14) = 2.976843
99% CI = (np.float64(1.0901973384384906), np.float64(1.3871359948948425))

--- Confidence Interval using KNOWN σ = 0.2 (z-distribution) ---
Population σ = 0.2
z critical = 2.575829
99% CI = (np.float64(1.1056514133957607), np.float64(1.3716819199375725))


**Conclusion:**

We are 99% confident that the true mean durability of the print heads lies between ~1.1 and ~1.4 million characters.

The two methods (t vs. z) give very similar ranges because the sample SD (≈0.193) is close to the assumed σ (0.2).

**Hypothesis Testing**

In [2]:
"""
Task:
- Test H0: mu = 4000 vs H1: mu > 4000 (one-sided, α=0.05)
- Weekly operating cost model: W = 1000 + 5X
- Given: X ~ N(600, 25^2), so σ_W = 125
- Sample: n = 25, sample mean = 3050
"""

import math
from scipy import stats

# Parameters
mu_X = 600
sigma_X = 25
mu0_W = 1000 + 5 * mu_X   # Theoretical mean
sigma_W = 5 * sigma_X     # Population SD for W
n = 25
xbar = 3050               # Sample mean
alpha = 0.05

# Compute test statistic
se_W = sigma_W / math.sqrt(n)
z_stat = (xbar - mu0_W) / se_W

# Critical value (one-sided, right-tail)
z_crit = stats.norm.ppf(1 - alpha)

# p-value (right-tail test)
p_value = 1 - stats.norm.cdf(z_stat)

print("\n--- Hypothesis Testing ---")
print(f"Null mean (mu0) = {mu0_W}")
print(f"Sample mean = {xbar}, n = {n}")
print(f"Population SD (σ_W) = {sigma_W}")
print(f"Standard error = {se_W:.6f}")
print(f"Test statistic z = {z_stat:.6f}")
print(f"Critical z (one-sided, α=0.05) = {z_crit:.6f}")
print(f"p-value (one-sided) = {p_value:.6e}")

# Decision
if z_stat > z_crit:
    print("Decision: Reject H0 (evidence mean > mu0)")
else:
    print("Decision: Fail to reject H0 (no evidence mean > mu0)")



--- Hypothesis Testing ---
Null mean (mu0) = 4000
Sample mean = 3050, n = 25
Population SD (σ_W) = 125
Standard error = 25.000000
Test statistic z = -38.000000
Critical z (one-sided, α=0.05) = 1.644854
p-value (one-sided) = 1.000000e+00
Decision: Fail to reject H0 (no evidence mean > mu0)


**Conclusion**

The sample mean is much lower than the theoretical mean of Rs. 4000, so there is no evidence that the true mean weekly cost is higher (as the restaurant owners claimed). In fact, the data suggest costs are significantly lower than the model expectation.